sys/include/suit.h

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2018 Freie Universit�t Berlin
+ * Copyright (C) 2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_suit SUIT manifest parser
+ * @ingroup     sys
+ * @brief       SUIT manifest parser
+ *
+ * This is a simple suit manifest parser for RIOT. The high level assumption is
+ * that the raw manifest data is stored in a buffered location where raw values
+ * or strings can be left during the lifetime of the suit_manifest_t struct.
+ * This assumption is valid in the case where a (network based) transfer
+ * mechanism is used to transfer the manifest to the node and an intermediate
+ * buffer is necessary to validate the manifest.
+ *
+ * The parser is based on draft version 1 of the specification.
+ *
+ * @see https://tools.ietf.org/html/draft-moran-suit-manifest-01
+ *
+ * @{
+ *
+ * @brief       Decoding functions for a SUIT manifest
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef SUIT_H
+#define SUIT_H
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
pls move this below the includes

Reply:
Out of interested (don't want to spark a discussion on this), is there a rule/guideline for the order of this somewhere?

Reply:
Changed btw

Reply:
> is there a rule/guideline for the order of this somewhere?

"do as we always did" combined with "close in reversed open order":

- copyright
- doxygen open (with file info)
- header guard open
- includes
- ifdef c++ open
- code
- ifdef c++ close
- header guard close
- doxygen close

for the includes it is somewhat important to be before the "extern C" as they *might* have other namespacing.

Reply:
Thanks for the clarification